packages/core/src/lib/button/icon-button.svelte

```diff
@@ -0,0 +1,73 @@
+<!--
+@component
+  
+For user triggered actions.
+
+```svelte
+<IconButton type="submit" variant="success" on:click={onClick} />
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import { createEventDispatcher } from 'svelte';
+import cx from 'classnames';
+import { Icon } from '$lib';
+
+/** The icon shown in the button. */
+export let icon: string;
+
+/** aria-label text for accessibility */
+export let label: string;
+
+/** Whether or not the button accepts clicks. */
+export let disabled = false;
+
+/** The behavior of the button. */
+export let type: 'button' | 'submit' | 'reset' = 'button';
+
+/** The communicated action type to the user. */
+export let variant: 'primary' | 'danger' = 'primary';
+
+/**
+ * The text that appears in a native popup box on hoveClassListArguements to the value
+ * of `label`.
+ */
+export let title = label;
+
+/** Additional CSS classes to pass to the button. */
+export let classes: cx.Argument = '';
```

Comment:
If we wanted to use `class` as the prop name, I think we can, according to the [Svelte docs on exporting props](https://svelte.dev/docs/svelte-components#script-1-export-creates-a-component-prop)!

To nitpick: `classes` feels a little non-standard and I'd rather use something conventional, like `class`. I even wonder if `cx` would make sense, to make it clear that it's gonna get passed to `classnames`

```ts
let extraClasses: cx.Argument = '';
$: classList = cx(..., extraClasses)

...

export { extraClasses as cx }
```

Reply:
Oh yeah I think we actually do that elsewhere so that would be great. I am not sure about `cx`, it kind of comes from the react world of `classSet` right? Maybe we just use `class` and at least the typing should show it can of `classnames` type? We can also call it out in documentation, but also just plain class strings work.